src/common/color/colors.ts

```diff
@@ -74,3 +75,12 @@ export function getGraphColorByIndex(
     getColorByIndex(index);
   return theme2hex(themeColor);
 }
+
+export const getAllGraphColors = memoizeOne(
+  (style: CSSStyleDeclaration) =>
+    COLORS.map((_color, index) => getGraphColorByIndex(index, style)),
+  (newArgs: [CSSStyleDeclaration], lastArgs: [CSSStyleDeclaration]) =>
+    // this is not ideal, but we need to memoize the colors
```

Comment:
Mmm this assumes the first color is changed in a theme, which isn't always the case, only the third color could be changed for example...

Reply:
Exactly. But checking all colors defeats the purpose. I haven't thought of a better solution yet

Reply:
To be fair, this only needs change when the theme is changed and there are custom colors, which is not that often. Plus a refresh would fix it even if the first color was the same